torch/testing/_internal/common_methods_invocations.py

```diff
@@ -601,6 +601,36 @@ def sample_inputs_linalg_vector_norm(op_info, device, dtype, requires_grad, **kw
 
     return inputs
 
+# These sample inputs are shared by add, sub & mul.
+def sample_inputs_add(op_info, device, dtype, requires_grad, **kwargs):
```

Comment:
Let's revisit this after the 'add' OpInfo is added. I'm guessing we'll want to restructure this so it's "sample_inputs_binary_pwise", which is a more general name if it's handling add/sub/mul/etc. Then we should refactor it slightly to support accepting kwargs for values like alpha so sample_inputs_add can just be a wrapper around sample_inputs_binary_pwise.

Reply:
Hello @mruberry, please confirm which other binary pointwise ops should share `sample_inputs_binary_pwise`. I'll refactor it based on your inputs. Thanks!

Reply:
Just sub and mul would be great for now. They're easy because they don't have an domain requirements (like div would).

Reply:
Thanks for the info, @mruberry!

Reply:
@mruberry, it looks like if we want to support alpha, beta kwargs, then we'd also have to conditionally pass `kwargs` to `SampleInput` by perhaps having another `kwarg` `supports_kwargs`,

For example, in a `SampleInput` call,
`kwargs=dict(alpha=alpha, beta=beta) if kwargs.get("supports_kwargs", False) else dict()`

Reply:
A couple options:

- we can not test add and sub's alpha for now
- when calling sample_inputs_binary_pwise OpInfos can partial `alpha` as a kwarg


```
OpInfo(foo...
  sample_inputs_func=partial(sample_inputs_binary_pwise, alpha=.5)...
```

then in sample_inputs_binary_pwise the alpha kwarg can be (optionally) fetched from `kwargs`. If it's present then a new SampleInput can be added that includes an `alpha` kwarg. This isn't great coverage of the `alpha` arguments for add or sub, but that's OK. We can extend it in the future if we want to.

Reply:
Thanks for the clarification!

I made changes with respect to the second option, but the only issue is that `alpha` shouldn't be float for integral types, as `add` & `sub` produce an error. So, I guess we can have at least two `alpha` kwargs - `alpha_floating` for floating types & `alpha_integral` for integral types. I think even `alpha_complex` is an option, as it can be a complex scalar.